feedback-module/src/components/pages/Screen2.js

```diff
@@ -8,34 +8,66 @@ import ModuleButton from "../common/Button";
 
 function Screen2({ feedbackType }) {
   const [checkedFields, setCheckedFields] = useState({});
+  const [otherField, setOtherField] = useState("");
+  const [inputQuestions, setInputQuestions] = useState({});
 
   useEffect(() => {
     let fields = {};
+
     feedbackType.checkboxes.map((label) => {
       fields = {
         ...fields,
         [label]: false,
       };
     });
     setCheckedFields(fields);
```

Comment:
I noticed that you also setCheckedFields in the onCheck and pass in setInputQuestions as a prop in TextBoxList— is it still necessary to have this useEffect?

Reply:
The onCheck function assumes that all the values are already in the checkedFields object (set to false) and the TextBoxList takes in the InputQuestions as a prop (also assuming it is already populated). This useEffect initially populates all the objects. I could remove it but it would require altering the other functions (I didn't think I would want to reset all the values every time a checkbox was checked or there was an onChange (I think I only need to set this once which is why I thought of putting it in the useEffect)

Reply:
Hmmm would it be possible to populate the objects when setting useState since we're only trying to do this for the initial render?

So instate of useState({}) we could maybe do something like
```
useState(
{feedbackType.checkboxes.map((label) => {
      fields = {
        ...fields,
        [label]: false,
      };
    });}
)
```

Reply:
Yeah I think a useEffect with an empty dependency array has the same effect but if you think this is cleaner then I can change it

Reply:
Not sure how I would populate the array inside the useState